blockchain/stake/staketx.go

```diff
@@ -1240,21 +1240,31 @@ func IsStakeSubmissionTxOut(index int) bool {
 //
 // The tickets are determined **only** from the STransactions of the provided
 // block and no validation is performed.
+//
+// This function is only safe to be called with a block that has
+// passed all header commitments.
```

Comment:
I would suggest a slightly different wording here since I think "passed all header commitments" is a bit ambiguous.

How about the following?

s/passed all header commitments/had all header commitments validated./